src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.OverlappedValueTaskSource.Windows.cs

```diff
@@ -195,34 +195,33 @@ private static void IOCallback(uint errorCode, uint numBytes, NativeOverlapped*
 
             internal void Complete(uint errorCode, uint numBytes)
             {
-                Debug.Assert(errorCode == Interop.Errors.ERROR_SUCCESS || numBytes == 0, $"Callback returned {errorCode} error and {numBytes} bytes");
-
                 OSFileStreamStrategy? strategy = _strategy;
                 ReleaseResources();
 
-                if (strategy is not null && _bufferSize != numBytes) // true only for incomplete operations
-                {
-                    strategy.OnIncompleteOperation(_bufferSize, (int)numBytes);
-                }
-
                 switch (errorCode)
                 {
                     case Interop.Errors.ERROR_SUCCESS:
                     case Interop.Errors.ERROR_BROKEN_PIPE:
                     case Interop.Errors.ERROR_NO_DATA:
                     case Interop.Errors.ERROR_HANDLE_EOF: // logically success with 0 bytes read (read at end of file)
+                        if (_bufferSize != numBytes) // true only for incomplete operations
+                        {
+                            strategy?.OnIncompleteOperation(_bufferSize, (int)numBytes);
+                        }
                         // Success
                         _source.SetResult((int)numBytes);
                         break;
 
                     case Interop.Errors.ERROR_OPERATION_ABORTED:
+                        strategy?.OnIncompleteOperation(_bufferSize, 0); // don't use numBytes here, as it can be != 0 for this errorCode (#57212)
```

Comment:
My usual comment that #57212 can immediately be found by git blame here.

Reply:
(but don't feel obliged to change it.)

Reply:
PR #57212 was opened to test Adam's hypothesis, which he then closed when it was confirmed. Unless I'm missing something, git blame only shows merged PRs.

Reply:
Ah true. Nevertheless, my personal preference would be to document whatever we learned right here (actually I think that's already done)

I know I'm a bit odd (although I think @stephentoub has the same general preference, we aren't consistent) and don't feel obliged to do anything here. We ought to reach collective consensus at some future point...